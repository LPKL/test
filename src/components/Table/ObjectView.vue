<template>
  <div :style="'width:'+width+';height:'+height+';display: inline-block;overflow-y: scroll;position: relative;'">
    <div style="position: absolute;bottom: 6px;width: 100%;">
      <el-row v-for="(item,key) in showData" :key="key" class="tablerow">
        <el-col :span="6"><div style="border-right: 1px solid #ccc;padding:10px 5px;">{{ key }}</div></el-col>
        <el-col :span="18"><div style="padding: 10px 5px;">{{ item.toString() }}</div></el-col>
      </el-row>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ObjectView',
  props: {
    showData: {
      type: Object,
      default: null
    },
    width: {
      type: String,
      default: '400px'
    },
    height: {
      type: String,
      default: '400px'
    }
  },
  data() {
    return {
      testdata: {}
    }
  }
}
</script>

<style rel="stylesheet/scss" lang="scss" scoped>
  .tablerow:first-child{
    border-top: 1px solid #ccc;
  }
  .tablerow{
    border-right: 1px solid #ccc;
    border-bottom: 1px solid #ccc;
    .el-col:first-child{
      border-left: 1px solid #ccc;
    }

  }
</style>
