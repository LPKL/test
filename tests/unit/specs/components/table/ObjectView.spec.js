/**
 * 完整测试示例
 * ObjectView测试用例完成
 */
import { shallowMount } from '@vue/test-utils'
import mountOption from '~/cores/vueinstance'
import ObjectView from '@/components/Table/ObjectView.vue'

describe('ObjectView.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(ObjectView, mountOption({
      showData: {
        name1: 'abc',
        name2: '3dtest'
      }}))
  })
  it('简单测试', () => {
    const vm = wrapper.vm // 利用vm测试组件内属性和方法
    vm.$nextTick(() => {
      expect(vm.$props.showData.name1).toBe('abc')
      expect(typeof wrapper.find).toBe('function')
      const tablerow = wrapper.findAll('.tablerow div') // 测试dom属性
      expect(tablerow.at(3).text()).toBe('3dtest') // 找到最后一个div对比他的文本是否渲染正确
    })
  })
  it('生成快照', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
