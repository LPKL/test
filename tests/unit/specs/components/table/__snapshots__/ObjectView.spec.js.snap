// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectView.vue 生成快照 1`] = `
<div
  style="width: 400px; height: 400px; display: inline-block; overflow-y: scroll; position: relative;"
>
  <div
    style="position: absolute; bottom: 6px; width: 100%;"
  >
    <el-row-stub
      align="top"
      class="tablerow"
      justify="start"
      tag="div"
    >
      <el-col-stub
        span="6"
        tag="div"
      >
        <div
          style="border-right: 1px solid #ccc; padding: 10px 5px;"
        >
          name1
        </div>
      </el-col-stub>
       
      <el-col-stub
        span="18"
        tag="div"
      >
        <div
          style="padding: 10px 5px;"
        >
          abc
        </div>
      </el-col-stub>
    </el-row-stub>
    <el-row-stub
      align="top"
      class="tablerow"
      justify="start"
      tag="div"
    >
      <el-col-stub
        span="6"
        tag="div"
      >
        <div
          style="border-right: 1px solid #ccc; padding: 10px 5px;"
        >
          name2
        </div>
      </el-col-stub>
       
      <el-col-stub
        span="18"
        tag="div"
      >
        <div
          style="padding: 10px 5px;"
        >
          3dtest
        </div>
      </el-col-stub>
    </el-row-stub>
  </div>
</div>
`;
